pdf/model/appender.go

```diff
@@ -370,20 +380,83 @@ func (a *PdfAppender) ReplacePage(pageNum int, page *PdfPage) {
 	}
 }
 
-// ReplaceAcroForm replaces the acrobat form. It appends a new form to the Pdf which replaces the original acrobat form.
+// Sign signs a specific page with a digital signature using a specified signature handler.
+// Returns an Acroform and PdfAppearance that can be used to customize the signature appearance.
+func (a *PdfAppender) Sign(pageNum int, handler SignatureHandler) (acroForm *PdfAcroForm, appearance *PdfAppearance, err error) {
+	acroForm = a.Reader.AcroForm
+	if acroForm == nil {
+		acroForm = NewPdfAcroForm()
+	}
+	pageIndex := pageNum - 1
```

Comment:
This can be done without looping through the pages slices. Something like:

```
pageIndex := pageNum - 1
if pageIndex < 0 || pageIndex > len(a.pages)-1 {
        return nil, nil, fmt.Errorf("page %d not found", pageNum)
}
page := a.pages[pageIndex].Duplicate()
```